packages/runtime-core/src/componentEmits.ts

```diff
@@ -92,10 +92,17 @@ export function emit(
             event.startsWith(compatModelEventPrefix))
         )
       ) {
-        if (!propsOptions || !(toHandlerKey(event) in propsOptions)) {
+        if (
+          !propsOptions ||
+          !(
+            // only checking on camelized handler keys since that is what developers should use
+            // https://github.com/vuejs/vue-next/pull/4804#discussion_r736462241
+            toHandlerKey(camelize(event)) in propsOptions
+          )
+        ) {
           warn(
-            `Component emitted event "${event}" but it is neither declared in ` +
-              `the emits option nor as an "${toHandlerKey(event)}" prop.`
+            `Component emitted event "${event}" but didn't declare a "${toHandlerKey(camelize(event))}" prop ` +
+              `or a "${camelize(event)}" emit option.`
```

Comment:
I understand that this wording came from https://github.com/vuejs/core/pull/4804#discussion_r1026231929, but I think the previous wording (with just the addition of `camelize` to the prop name) was better. In particular:

1. The suggestion to use a prop should come last, as using `emits` is a much more common solution than using a prop.
2. The suggested name for the `emits` option in this warning is currently incorrect. e.g. See [Playground](https://deploy-preview-8268--vue-sfc-playground.netlify.app/#__DEV__eNo9jMEKwjAQRH9lWYS2B81doiiiP2E8aLqFYJssZuNF/HfXtvQ4M2/eB4/Mm3ch3KLN/hVYIJMU3rvYUhcinYcgub5Whdu70CWl6ta4aM0EK6ZBaOBeV00A9lFEUoSD74N/7hyuSBX1LFh3amgcjijA6c+MLzPdtLdm8eH3B/o6N9Q=), which claims that `updateFoo` should be added to `emits`, even though it is already included. The check for `event in emitsOptions` a few lines up isn't using `camelize`, so the warning needs to be consistent with that.